packages/tables/resources/views/columns/toggle-column.blade.php

```diff
@@ -2,135 +2,116 @@
     $state = $getState();
 @endphp
 
-<div
-    x-data="{
-        error: undefined,
-        state: @js((bool) $state),
-        isLoading: false,
-    }"
-    x-init="
-        $watch('state', () => $refs.button?.dispatchEvent(new Event('change')))
-
-        Livewire.hook('message.processed', (component) => {
-            if (component.component.id !== @js($this->id)) {
-                return
-            }
-
-            if (! $refs.newState) {
-                return
-            }
-
-            let newState = $refs.newState.value === '1' ? true : false
+<div wire:key="{{ $this->id }}.table.record.{{ $recordKey }}.column.{{ $getName() }}.toggle-column.{{ json_encode($state) }}">
```

Comment:
Not sure about having the state in the key.

Reply:
You need the state in the key so that when it changes on the edit action it will rerender the toggle component.

Reply:
Well, with it rerenders each time it's toggled, but this allows it to work with the edit action. 

And that combined with the wire:ignore on the child div prevents the duplicate queries when filtering.

Reply:
IMO this is a good fix for v2 since it solves all the current issues with toggle columns, however, I think this can be revisted again when livewire v3 ships. I still think there's some bug with alpine/livewire with the combination of init and the watcher that's causing these dom diffing problems. Maybe those will be fixed with v3 and we can revisit and get the transition working again when toggling to `off`.

Reply:
Pushed another update to repo for #6215 that overrides the toggle-column with this new code so you don't have to copy and paste anymore. Hope that helps 👍

Reply:
Do we even need the state in the wire:key if it contains the record and column keys/names already?

Reply:
Yes, you need the state in the key so that when the state changes on toggle it will rerender the toggle component. This allows us to get rid off all init and $watch and hidden input which were causing the issues.